sshconnect2.c

```diff
@@ -1262,10 +1262,20 @@ identity_sign(struct identity *id, u_char **sigp, size_t *lenp,
 			goto out;
 		}
 		sign_key = prv;
-		if (sshkey_is_sk(sign_key)) {
+		if (sshkey_is_sk(sign_key) &&
+		    (sign_key->sk_flags & SSH_SK_USER_PRESENCE_REQD)) {
+			fmprintf(stdout, "You may need to touch your "
+				 "authenticator.\n");
```

Comment:
Maybe move the `notify_start(..., "Confirm user presence...")` code here?

Reply:
Hmm. I think I did that during hacking this stuff and somehow it didn't quite work as desired... but ATM I don't remember.
Let me check.

Reply:
Ok, the problem is that the notification prompt is only required if we're not using WinHello, so we may only emit the notification if the key is not accepted by the first call of sshkey_sign with SSH_ERR_KEY_WRONG_PASSPHRASE, which automatically implies we're not using WinHello. The printout of "You may need to touch your authenticator." is an uninformed hint to the user at this point and not the start of actually requesting user presence.

Reply:
Maybe wrap it in `#ifdef CYGWIN`? AFAIK it will produce notifications that are redundant to notify_start() on other platforms. It would be nicer if we could plumb the is_winhello() thing through, but I think that would require an API change to the middleware interface...

Reply:
Yeah, the missing info about the device type on the upper level is a bit of a problem, but nothing we can't cope with.

I can wrap that in a #if, but I was trying to avoid conditional code. Can you check this on another OS to see how this looks like on non-Windows platforms? Maybe it's not that bad...?

Reply:
If it's non-conditional, then I'd prefer the notification be done using `notify_start()`, as that is the way that all the other presence notifications are handled. In particular it respects the user's `$SSH_ASKPASS_REQUIRE` setting